### ISSUE TITLE
React Native 0.74 Support

### ISSUE THREAD

**Issue by tsopin** [klaviyo/klaviyo-react-native-sdk#150]
### Checklist

- [X] I have read the [contributing guidelines](https://github.com/klaviyo/klaviyo-react-native-sdk/blob/master/.github/CONTRIBUTING.md)
- [X] I have determined whether this bug is also reproducible in a vanilla project, such as the example app in this repository.
- [X] If possible, I've reproduced the issue using the `master` branch or latest release of this package.
- [X] This issue hasn't been addressed in an [existing issue](https://github.com/klaviyo/klaviyo-react-native-sdk/issues) or [pull request](https://github.com/klaviyo/klaviyo-react-native-sdk/pulls)

### Description

Installing `klaviyo-react-native-sdk@0.4.0` in iOS fails with the latest version of React Native 0.74.

### Expected behavior

Successful installation

### Actual behavior

When running `pod install` on a newly created RN 0.74 project, it fails with: 

```markdown
[!] Invalid `Podfile` file:
...
ModuleInterfaceNotFoundParserError: Module NativeKlaviyoReactNativeSdk: No TypeScript interfaces extending TurboModule were detected in this NativeModule spec.
...
```

### Steps to reproduce

```markdown
1. Create a new React Native project `npx react-native@latest init AwesomeProject`
2. Install Klaviyo `yarn add klaviyo-react-native-sdk`
3. Run `pod install` from `ios` folder of the project
```

### The Klaviyo React Native SDK version information

0.4.0

### Environment Description

System:   OS: macOS 14.3   CPU: (12) arm64 Apple M2 Max   Memory: 434.53 MB / 32.00 GB   Shell:     version: "5.9"     path: /bin/zsh Binaries:   Node:     version: 21.4.0     path: /opt/homebrew/bin/node   Yarn:     version: 1.22.19     path: /usr/local/bin/yarn   npm:     version: 10.2.4     path: /opt/homebrew/bin/npm   Watchman:     version: 2023.12.04.00     path: /opt/homebrew/bin/watchman Managers:   CocoaPods:     version: 1.15.2     path: /opt/homebrew/bin/pod SDKs:   iOS SDK:     Platforms:       - DriverKit 23.0       - iOS 17.0       - macOS 14.0       - tvOS 17.0       - watchOS 10.0   Android SDK:     API Levels:       - "33"       - "34"     Build Tools:       - 30.0.3       - 33.0.0       - 33.0.1       - 33.0.2       - 34.0.0     System Images:       - android-33 | Android TV ARM 64 v8a       - android-33 | Google TV ARM 64 v8a       - android-33 | Google APIs ARM 64 v8a       - android-33 | Google Play ARM 64 v8a       - android-34 | Google APIs ARM 64 v8a     Android NDK: Not Found IDEs:   Android Studio: 2022.3 AI-223.8836.35.2231.10671973   Xcode:     version: 15.0.1/15A507     path: /usr/bin/xcodebuild Languages:   Java:     version: 17.0.8.1     path: /usr/bin/javac   Ruby:     version: 2.6.10     path: /usr/bin/ruby npmPackages:   "@react-native-community/cli": Not Found   react:     installed: 18.2.0     wanted: 18.2.0   react-native:     installed: 0.74.0     wanted: 0.74.0   react-native-macos: Not Found npmGlobalPackages:   "*react-native*": Not Found Android:   hermesEnabled: true   newArchEnabled: false iOS:   hermesEnabled: true   newArchEnabled: false

**Comment by ajaysubra** [comment]
Hey @tsopin Thanks for reporting this issue to us and providing all the details. We are able to reproduce this on our end and will get back to you as soon as we have a solution.

**Comment by AronBe** [comment]
Build worked for me 👍

**Comment by ajaysubra** [comment]
Thanks for confirming @AronBe 

@tsopin this should be fixed now and available in the latest release [`0.4.1`](https://github.com/klaviyo/klaviyo-react-native-sdk/releases/tag/0.4.1). Thanks for your patience.
